### PR TITLE
[REVIEW] Pin `dask` and `distributed` for release

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR pins `dask` and `distributed` to `2022.9.2` for `22.10` release.

xref: https://github.com/rapidsai/cudf/pull/11822

### CODE REVIEW

**Review by raydouglass — APPROVED** [review]

**Comment by galipremsagar** [comment]
@charlesbluca looks like the mix of `dask-core`(nightly) & `dask`(conda-forge) packages is biting us here again:
```python
(cudfdev) pgali@dt07:/nvme/0/pgali/dask-cuda$ python
Python 3.9.13 | packaged by conda-forge | (main, May 27 2022, 16:56:21) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import dask_cuda
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nvme/0/pgali/dask-cuda/dask_cuda/__init__.py", line 12, in <module>
    from .cuda_worker import CUDAWorker
  File "/nvme/0/pgali/dask-cuda/dask_cuda/cuda_worker.py", line 12, in <module>
    from distributed import Nanny
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.9/site-packages/distributed/__init__.py", line 30, in <module>
    from distributed.deploy import Adaptive, LocalCluster, SpecCluster, SSHCluster
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.9/site-packages/distributed/deploy/__init__.py", line 7, in <module>
    from distributed.deploy.local import LocalCluster
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.9/site-packages/distributed/deploy/local.py", line 13, in <module>
    from distributed.deploy.utils import nprocesses_nthreads
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.9/site-packages/distributed/deploy/utils.py", line 6, in <module>
    from dask.utils import factors
ImportError: cannot import name 'factors' from 'dask.utils' (/nvme/0/pgali/envs/cudfdev/lib/python3.9/site-packages/dask/utils.py)
```

So I went ahead and added a flag-based channel(`dask/label/dev`) removal.

cc: @jakirkham @pentschev

**Comment by pentschev** [comment]
Regarding the pinned version, it seems like there will be a release tomorrow after all: https://github.com/dask/community/issues/278 . If possible, let's pin to that, we want to get https://github.com/dask/distributed/pull/6996 in which will fix all the uncatched exceptions when shutting down a Dask cluster with UCX.

File: ci/cpu/build.sh [hunk 1]
```diff
@@ -19,6 +19,10 @@ export CUDA_REL=${CUDA_VERSION%.*}
 export GPUCI_CONDA_RETRY_MAX=1
 export GPUCI_CONDA_RETRY_SLEEP=30
 
+# Whethere to keep `dask/label/dev` channel in the env. If INSTALL_DASK_MAIN=0,
```
Comment (galipremsagar) [inline ci/cpu/build.sh:22]:
```suggestion
# Whether to keep `dask/label/dev` channel in the env. If INSTALL_DASK_MAIN=0,
```

File: ci/cpu/build.sh [hunk 2]
```diff
@@ -61,8 +71,8 @@ conda list --show-channel-urls
 # FIX Added to deal with Anancoda SSL verification issues during conda builds
 conda config --set ssl_verify False
 
-pip install git+https://github.com/dask/dask.git@main
-pip install git+https://github.com/dask/distributed.git@main
+pip install git+https://github.com/dask/dask.git@2022.9.1
+pip install git+https://github.com/dask/distributed.git@2022.9.1
```
Comment (galipremsagar) [inline ci/cpu/build.sh:75]:
```suggestion
pip install git+https://github.com/dask/dask.git@2022.9.2
pip install git+https://github.com/dask/distributed.git@2022.9.2
```

File: ci/gpu/build.sh [hunk 3]
```diff
@@ -35,10 +35,10 @@ export NUMPY_EXPERIMENTAL_ARRAY_FUNCTION=1
 
 # Install dask and distributed from main branch. Usually needed during
 # development time and disabled before a new dask-cuda release.
-export INSTALL_DASK_MAIN=1
+export INSTALL_DASK_MAIN=0
 
 # Dask version to install when `INSTALL_DASK_MAIN=0`
-export DASK_STABLE_VERSION="2022.7.1"
+export DASK_STABLE_VERSION="2022.9.1"
```
Comment (galipremsagar) [inline ci/gpu/build.sh:41]:
```suggestion
export DASK_STABLE_VERSION="2022.9.2"
```

File: requirements.txt [hunk 4]
```diff
@@ -1,5 +1,5 @@
-dask>=2022.7.1
-distributed>=2022.7.1
+dask==2022.9.1
+distributed==2022.9.1
```
Comment (galipremsagar) [inline requirements.txt:2]:
```suggestion
dask==2022.9.2
distributed==2022.9.2
```

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

File: ci/cpu/build.sh [hunk 5]
```diff
@@ -43,9 +47,15 @@ gpuci_logger "Activate conda env"
...
 
-# Remove rapidsai-nightly channel if we are building main branch
+# Remove `rapidsai-nightly` & `dask/label/dev` channel if we are building main branch
 if [ "$SOURCE_BRANCH" = "main" ]; then
   conda config --system --remove channels rapidsai-nightly
+  conda config --system --remove channels dask/label/dev
+fi
+
+# Remove `dask/label/dev` channel if INSTALL_DASK_MAIN=0
+if [[ "${INSTALL_DASK_MAIN}" == 0 ]]; then
```
Comment (galipremsagar) [inline ci/cpu/build.sh:57]:
```suggestion
elif [[ "${INSTALL_DASK_MAIN}" == 0 ]]; then
# Remove `dask/label/dev` channel if INSTALL_DASK_MAIN=0
```

File: ci/cpu/build.sh [hunk 6]
```diff
@@ -61,8 +69,8 @@ conda list --show-channel-urls
 # FIX Added to deal with Anancoda SSL verification issues during conda builds
 conda config --set ssl_verify False
 
-pip install git+https://github.com/dask/dask.git@main
-pip install git+https://github.com/dask/distributed.git@main
+pip install git+https://github.com/dask/dask.git@2022.9.2
+pip install git+https://github.com/dask/distributed.git@2022.9.2
```
Comment (pentschev) [inline ci/cpu/build.sh:73]: Maybe in a follow-up PR, but would it make more sense to use the same logic as we do in `ci/gpu/build.sh`, or is there a reason I'm not aware for keeping `pip install`?

**Review by pentschev — APPROVED** [review]
LGTM, thanks @galipremsagar !

File: ci/cpu/build.sh (hunk 6 above)
Comment (pentschev) [reply ci/cpu/build.sh:73]: Ah ok, that would be even better. Thanks Prem!

**Comment by galipremsagar** [comment]
@gpucibot merge
@gpucibot merge

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**